crates/sdk/tests/integration_test.rs

```diff
@@ -580,11 +585,109 @@ fn test_segmentation_retry() {
     let sdk = Sdk::new();
-    sdk.generate_root_verifier_asm(&agg_pk);
+    let guest_opts = GuestOptions::default();
+    let mut pkg_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).to_path_buf();
+    pkg_dir.push("guest/fib");
+    let elf = sdk.build(guest_opts.clone(), &pkg_dir, &Default::default())?;
+
+    let vm_config = SdkVmConfig::builder()
+        .system(SdkSystemConfig {
+            config: SystemConfig::default().with_continuations(),
+        })
+        .rv32i(Default::default())
+        .rv32m(Default::default())
+        .io(Default::default())
+        .native(Default::default())
+        .build();
+    assert!(vm_config.system.config.continuation_enabled);
+
+    let app_exe = sdk.transpile(elf, vm_config.transpiler())?;
+    let fri_params = FriParameters::new_for_testing(LEAF_LOG_BLOWUP);
+    let app_config = AppConfig::new_with_leaf_fri_params(fri_params, vm_config.clone(), fri_params);
+
+    let app_pk = sdk.app_keygen(app_config.clone())?;
+    let committed_app_exe = sdk.commit_app_exe(fri_params, app_exe.clone())?;
+
+    let commits =
+        AppExecutionCommit::compute(&vm_config, &committed_app_exe, &app_pk.leaf_committed_exe);
+
+    let agg_pk = AggStarkProvingKey::keygen(AggStarkConfig {
+        max_num_user_public_values: DEFAULT_MAX_NUM_PUBLIC_VALUES,
+        leaf_fri_params: FriParameters::new_for_testing(LEAF_LOG_BLOWUP),
+        internal_fri_params: FriParameters::new_for_testing(INTERNAL_LOG_BLOWUP),
+        root_fri_params: FriParameters::new_for_testing(ROOT_LOG_BLOWUP),
+        profiling: false,
+        compiler_options: CompilerOptions {
+            enable_cycle_tracker: true,
+            ..Default::default()
+        },
+        root_max_constraint_degree: (1 << ROOT_LOG_BLOWUP) + 1,
+    });
+    let asm = sdk.generate_root_verifier_asm(&agg_pk);
+    let asm_path = format!(
+        "{}/guest/verify_openvm_stark/{}",
+        env!("CARGO_MANIFEST_DIR"),
+        ASM_FILENAME
+    );
+    std::fs::write(asm_path, asm)?;
+
+    let e2e_stark_proof = sdk.generate_e2e_stark_proof(
+        Arc::new(app_pk),
+        committed_app_exe,
+        agg_pk,
+        StdIn::default(),
+    )?;
+
+    let verify_exe = {
+        let mut pkg_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).to_path_buf();
+        pkg_dir.push("guest/verify_openvm_stark");
+        let elf = sdk.build(guest_opts.clone(), &pkg_dir, &Default::default())?;
+        sdk.transpile(elf, vm_config.transpiler())?
+    };
+
+    let pvs = [13u32, 21, 0, 0, 0, 0, 0, 0];
+
+    let exe_commit_u32: Vec<_> = commits
+        .exe_commit
+        .iter()
+        .map(|x| x.as_canonical_u32())
+        .collect();
+    let vm_commit_u32: Vec<_> = commits
+        .leaf_vm_verifier_commit
+        .iter()
+        .map(|x| x.as_canonical_u32())
+        .collect();
+    let pvs_u32: Vec<_> = pvs
+        .iter()
+        .flat_map(|x| x.to_le_bytes())
+        .map(|x| x as u32)
+        .collect();
+
+    let key = ASM_FILENAME
```

Comment:
why is the filename part of the key?
and if it is part of the key, why the base filename and not the full path?